src/Tests/AzureClientMagicTests.cs

```diff
@@ -232,6 +239,14 @@ public void TestSubmitMagic()
             submitMagic.Test($"{operationName}");
             Assert.AreEqual(AzureClientAction.SubmitJob, azureClient.LastAction);
             Assert.IsTrue(azureClient.SubmittedJobs.Contains(operationName));
+
+            // jobParams argument
+            Assert.IsTrue(azureClient.JobParams.IsEmpty);
+            submitMagic.Test($"{operationName} jobParams={jobParams}");
+            Assert.IsTrue(azureClient.JobParams.TryGetValue(jobParamsKey1, out string value1));
+            Assert.AreEqual(value1, jobParamsVal1);
```

Comment:
This pattern happens enough that it may be worth adding a new `HasKey` extension method to the `Assert.That` object. Roughly,

```csharp
void HasKey<TKey, TValue>(this Assert assert, IDictionary<TKey, TValue> dict, TKey key, TValue expected)
{
     Assert.IsTrue(dict.TryGetValue(key, out TValue actual));
     Assert.AreEqual(actual, expected);
}
```

Reply:
In my humble opinion, that extra layer of abstraction would make this code a little harder to read.  Granted, there is a bit of repetition, but not enough to significantly complicate maintenance.

Reply:
Fair enough; I've found such abstractions to be highly useful in the code base so far, in part because it reduces friction to writing new tests that would otherwise introduce more duplication. This abstraction also reduces the probability of forgetting to check that the key exists when writing new tests.

That said, it is a minor comment, and something easily handled in code maintenance following this PR regardless.

Reply:
Thanks a lot @cgranade ; for the purposes of this change let's leave this as is. I'll mark the comment as resolved. Going forward, it it makes sense to abstract this as part of feature work, let's do it. :)